feat: document how to install the docs

honzajavorek opened the pull request:
Documents @mnmkng's approach as mentioned in https://github.com/apify/apify-docs/issues/909#issuecomment-2068960909 and closes https://github.com/apify/apify-docs/issues/909. If https://github.com/apify/apify-docs/issues/936 gets some responses, I can add commits rewording some of the sentences too, but as of now I leave that part as it is.

honzajavorek commented:
The `request_queue.md` contains a link to a non-existing part of the Academy 🤔 

> A detailed tutorial on how to crawl one request queue from multiple Actor runs can be found in [Advanced web scraping academy](https://docs.apify.com/academy/advanced-web-scraping/multiple-runs-scrape).

Looks like the page links https://github.com/apify/apify-docs/pull/918, which isn't merged yet? And it fails under my Pull Request…

honzajavorek commented on CONTRIBUTING.md:
```diff
@@ -79,13 +79,35 @@ For example:
 Over time, we should track which items are useful and which don't get any traffic. Also, as Apify Docs expand, we may need to add more cards and update which articles we link to.
 
-## Local setup
+## Installation and setup
```
@B4nan I'd appreciate if you could take a look whether I got these sections right.

B4nan replied:
sure thing

TC-MO commented:
Yeah, you can disregard the failing lychee check, as it is pertaining only to the soon released tutorial.

B4nan commented on CONTRIBUTING.md:
```diff
@@ -79,13 +79,35 @@ For example:
+3. Run `npm install`.
+4. Run `npm start`. The website should open at <http://localhost:3000>.
+
+This is sufficient to work on the main content, i.e. the Academy and Platform docs.
+If you want to also work on the other parts of the docs, you may want to first join them all together using Nginx.
```
nit: we dont use prose wrap in markdown, its 2024 and editors can wrap things for us properly if we want them to do so (and its irrelevant for the rendered version). this is actually making things worse when reviewing the git diff (in general, its not relevant to this particular place).

at some point i'd like to add some code formatter that will deal with this automatically, we actually have/had a prettier setup here but it was more of a one time thing during the big migration. maybe we could prioritize that since there are now two people working primarily on the docs, it should be simple to revive. we are experimenting with biome which is super fast so it would be even suitable to have that enable on save

honzajavorek replied:
I do it for [better writing](https://sive.rs/1s), not so much for better git diffs. I'll do whatever you folks prefer, because as you say, it's two different things how I type characters to the file vs how they're committed to git.

Two people is a low number, I think we can agree on a convention just fine without tools policing us. But from what I've seen just for the past week, there's many people contributing to docs. If I'm the only one doing this tho, I still think it's easier to just tell me not to do it 😀

If you ever set up automation, I'll look over your shoulder 😀 I have Prettier on junior.guru, but I had to turn off formatting for many Markdown documents, because I have them mixed with Jinja2 etc., and that isn't something Prettier would understand.

honzajavorek replied:
Fixed in [35f8595](https://github.com/apify/apify-docs/pull/937/commits/35f8595dc454d95b5f5b06bc2af4eea894775c9a)

B4nan commented on CONTRIBUTING.md:
````diff
@@ -125,12 +145,14 @@ server {
 ```
 
-And add a record to `/etc/hosts` to map the `docs.apify.loc` hostname to localhost:
+Add a record to `/etc/hosts`, which maps the `docs.apify.loc` host to a localhost:
````
nit: `hostname` was probably more correct, from what I read, `host` is also supposed to contain the port, while `hostname` is just the name. but maybe not, haven't been googling that hard (and was happy with the first result that confirmed what I thought)

B4nan reviewed (approved):
just a few notes, nothing blocking

honzajavorek commented on CONTRIBUTING.md:
```diff
@@ -79,13 +79,35 @@ For example:
 
-If you want to work only on the main documentation content, cloning this repository is enough. Once you run `npm install` to install all dependencies and run `npm start`, the main portal will open on <http://localhost:3000>. All the links in the navbar and footer need to be absolute, and they will use a different hostname, configured to `docs.apify.loc` - to use that, follow the steps below and s ...
+1. Use Git to clone this repository.
+2. If you work at Apify and can install packages prefixed with `@apify-packages`, you can skip this step.
```
Fixed in [1c086c6](https://github.com/apify/apify-docs/pull/937/commits/1c086c636cc34f23b2cf1b012b23b0752667967f)

honzajavorek commented:
Even though they were not blocking, I addressed the comments. I'll merge this now, but if you find anything to correct, feel free to comment further - I can send fixes in a subsequent PR.
